docs/book/src/plugins/external-plugins.md

```diff
@@ -0,0 +1,155 @@
+	APIVersion string `json:"apiVersion"`
+
+	// Args holds the plugin specific arguments that are received from the CLI
+	// which are to be passed down to the external plugin.
+	Args []string `json:"args"`
+
+	// Command contains the command to be executed by the plugin such as init, create api, etc.
+	Command string `json:"command"`
+
+	// Universe represents the modified file contents that gets updated over a series of plugin runs
+	// across the plugin chain. Initially, it starts out as empty.
+	Universe map[string]string `json:"universe"`
+}
+```
+
+`PluginResponse` is what the plugin constructs with the updated universe and serialized and sent back to Kubebuilder through `stdout`. 
+
+```go
+// PluginResponse is returned to kubebuilder by the plugin and contains all files
+// written by the plugin following a certain command.
+type PluginResponse struct {
+	// APIVersion defines the versioned schema of the PluginResponse that is back sent back to Kubebuilder.
+	// Initially, this will be marked as alpha (v1alpha1)
+	APIVersion string `json:"apiVersion"`
+
+	// Command holds the command that gets executed by the plugin such as init, create api, etc.
+	Command string `json:"command"`
+
+	// Metadata contains the plugin specific help text that the plugin returns to Kubebuilder when it receives
+	// `--help` flag from Kubebuilder.
+	Metadata plugin.SubcommandMetadata `json:"metadata"`
+
+	// Universe in the PluginResponse represents the updated file contents that was written by the plugin.
+	Universe map[string]string `json:"universe"`
+
+	// Error is a boolean type that indicates whether there were any errors due to plugin failures.
+	Error bool `json:"error,omitempty"`
+
+	// ErrorMsgs contains the specific error messages of the plugin failures.
+	ErrorMsgs []string `json:"errorMsgs,omitempty"`
+
+	// Flags contains the plugin specific flags that the plugin returns to Kubebuilder when it receives
+	// a request for a list of supported flags from Kubebuilder
+	Flags []Flag `json:"flags,omitempty"`
+}
+```
+
+<aside class="note">
+<h1>Caution</h1>
+
+When writing your own external plugin, you **should not** directly echo or print anything to the stdout. 
+
+This is because Kubebuilder and your plugin are communicating with each other via `stdin` and `stdout` using structured `JSON` data.
+Any additional information sent to stdout (such as debug messages or logs) that's not part of the expected PluginResponse JSON structure may cause parsing errors when Kubebuilder tries to read and decode the response from your plugin.
+
+If you need to include logs or debug messages while developing your plugin, consider writing these messages to a log file instead.
+
+</aside>
+
+## How to use it?
+
+### Prerequisites
+
+- You have built and installed your Kubebuilder binary.
+
+- You have built and installed your external plugin as an **executable / binary**.
+
+- Your plugin is put in a path following the GVK schema:
+```sh
+/path1/path2/${name}/${version} 
```

Comment:
I like @camilamacedo86's suggestion.
> WDYT about something more like ^ ?

```suggestion
- kubebuilder CLI > 3.11.0
- Your own external plugin executable written in a language of your choice 
- Path of the external plugin configured via the ENV VAR (specify the environment variable here) or it can be placed in a path following a `group-like name and version` scheme:

```